tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge/file_io_processor.cpp

```diff
@@ -406,3 +407,42 @@ uint32_t FileSeek(std::string fileName, size_t position, uint8_t *result,
     return SUCCESS;
 }
 
+uint32_t FileDelete(string fileName, uint8_t *result, size_t resultSize) {
+    string resultStr;
+    size_t resultStrSize = 0;
+
+    if ( IsFileNameEmpty(fileName, result, resultSize) ) {
+        tcf::Log(TCF_LOG_ERROR, "Filename is empty\n");
+        return FAILED;
+    }
+
+    try {
+        if ( remove(fileName.c_str()) != 0 ) {
+            tcf::Log(TCF_LOG_DEBUG, "File %s delete failed", fileName.c_str());
+            resultStr = "FILE DELETE FAILED";
+            resultStrSize = resultStr.length();
+            result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultStrSize);
+            tcf::error::ThrowIf<tcf::error::ValueError>(
+                resultStrSize > resultSize, "result string size is "
+                "greater than max result size");
```

Comment:
We should check for error condition before strncpy and throw error

Reply:
Done